sled-agent/src/rack_setup/service.rs

```diff
@@ -151,26 +171,63 @@ impl RackSetupService {
     ///
     /// Arguments:
     /// - `log`: The logger.
-    /// - `config`: The config file, which is used to setup the rack.
-    /// - `peer_monitor`: The mechanism by which the setup service discovers
-    ///   bootstrap agents on nearby sleds.
+    /// - `request`: The requested rack setup
     /// - `local_bootstrap_agent`: Communication channel by which we can send
     ///   commands to our local bootstrap-agent (e.g., to initialize sled
     ///   agents).
     pub(crate) fn new(
         log: Logger,
-        config: Config,
-        local_bootstrap_agent: BootstrapAgentHandle,
+        request: RackInitializeRequest,
+        local_bootstrap_agent: impl BootstrapNetwork,
         // TODO-cleanup: We should be collecting the device ID certs of all
         // trust quorum members over the management network. Currently we don't
         // have a management network, so we hard-code the list of members and
         // accept it as a parameter instead.
         member_device_id_certs: Vec<Ed25519Certificate>,
     ) -> Self {
+        let interfaces = Interfaces {
+            dns: RealDnsAccess {},
+            nexus: RealNexusAccess {},
+            sleds: RealSledAccess {},
+            bootstrap_network: local_bootstrap_agent,
+        };
+
+        let marker_directory = omicron_common::OMICRON_CONFIG_PATH.into();
+        Self::new_internal(
+            log,
+            interfaces,
+            marker_directory,
+            request,
+            member_device_id_certs,
+        )
```

Comment:
This is basically the meat-and-potatoes of this PR -- everything else falls out of it.

This new interface shares the same structure as the old implementation, but all access to:
- DNS
- Nexus
- Sled agents
- The bootstrap agent
- Local storage

Is controlled explicitly by these input arguments, meaning that tests can define "whatever they want" and observe RSS behaves as expected under test.